pkg/store/tsdb.go

```diff
@@ -214,13 +220,24 @@ func (s *TSDBStore) LabelNames(ctx context.Context, r *storepb.LabelNamesRequest
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+
+	extLabelNames := make([]string, 0, len(s.extLset))
+	for _, lbl := range s.extLset {
+		extLabelNames = append(extLabelNames, lbl.Name)
+	}
+	res = strutil.MergeSlices(res, extLabelNames)
 	return &storepb.LabelNamesResponse{Names: res}, nil
 }
 
 // LabelValues returns all known label values for a given label name.
 func (s *TSDBStore) LabelValues(ctx context.Context, r *storepb.LabelValuesRequest) (
 	*storepb.LabelValuesResponse, error,
 ) {
+	withinTimeRange, err := s.checkRequestTimeRange(r.End)
```

Comment:
This is out of scope for this PR? Since it's small it's ok, let's just add this to PR tittle 🤗